### PR TITLE
secretscanner: init at 20210214-42a38f9

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Find secrets and passwords in container images and file systems

https://github.com/deepfence/SecretScanner

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113145` at 060cb635 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>secretscanner</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113145` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>secretscanner</li>
  </ul>
</details>

File: pkgs/tools/security/secretscanner/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    repo = "SecretScanner";
+    inherit rev;
+    sha256 = "0yga71f7bx5a3hj5agr88pd7j8jnxbwqm241fhrvv8ic4sx0mawg";
+  };
+
+  vendorSha256 = "0b7qa83iqnigihgwlqsxi28n7d9h0dk3wx1bqvhn4k01483cipsd";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  propagatedBuildInputs = [ hyperscan ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/secretscanner/default.nix:24]: If this is linked against it should be buildInputs.

Comment (fabaff) [reply pkgs/tools/security/secretscanner/default.nix]: It looks like that it's not needed during runtime. Thus only linked.

Thanks for the feedback.

File: pkgs/tools/security/secretscanner/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildGoModule
+, docker
```
Comment (SuperSandro2000) [inline pkgs/tools/security/secretscanner/default.nix:3]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113145` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>secretscanner</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

secretscanner:
<details>


Unused argument: `docker`.
Near pkgs/tools/security/secretscanner/default.nix:3:3:

```
  |
3 | , docker
  |   ^
```
</details>

File: pkgs/tools/security/secretscanner/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/security/secretscanner/default.nix]: Sorry for the extra round.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113145` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>secretscanner</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113145` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>secretscanner</li>
  </ul>
</details>

**Pull request merged**